src/System.Memory/src/System/ReadOnlySpanExtensions.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    /// <summary>
+    /// Extension methods for ReadOnlySpan&lt;T&gt;.
+    /// </summary>
+    public static class ReadOnlySpanExtensions
+    {
+        /// <summary>
+        /// Creates a new readonly span over the portion of the target string.
+        /// </summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is null.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ReadOnlySpan<char> Slice(this string text)
```

Comment:
This isn't implementable using the public surface area (especially w/ corefxlab still on C#6).

DangerousCreate() takes a "ref T" - corefxlab wanted it it to take an IntPtr.